### ISSUE TITLE
create_test --retry fails if the test is doing baseline generation

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#3912]
If you run `create_test --retry 1` together with the `--generate` option, the test fails to launch upon retry with an error like this:

```
ERROR: Baseline directories already exists ['/Users/sacks/projects/baselines/cimetest_0407a/SMS_D.f09_g16.X.bishorn_gnu']
Use -o to avoid this error
```

I think the fix is pretty simple, and I'll open a PR shortly.

Reproducer:

Introduce these diffs to force a run failure:

```diff
diff --git a/src/drivers/mct/main/cime_comp_mod.F90 b/src/drivers/mct/main/cime_comp_mod.F90
index 1611b00f8..a8a5537bf 100644
--- a/src/drivers/mct/main/cime_comp_mod.F90
+++ b/src/drivers/mct/main/cime_comp_mod.F90
@@ -695,6 +695,7 @@ subroutine cime_pre_init1(esmf_log_option)

     call mpi_init(ierr)
     call shr_mpi_chkerr(ierr,subname//' mpi_init')
+    call shr_sys_abort("Ending run here")

     end_count = shr_sys_irtc(irtc_rate)
     mpi_init_time = real( (end_count-beg_count), r8)/real(irtc_rate, r8)
```

then run create_test like this:

```
./create_test --retry 1 SMS_D.f09_g16.X.bishorn_gnu --generate cimetest_0407a
```

**Comment by ekluzek** [comment]
@billsacks is a bug or a feature that requires using the "--allow-baseline-overwrite" option? I imagine it works when the overwrite option right?  So does it just mean the two have to be used. together? I actually used to. use the overwrite option.,  but stopped when it didn't seem to always be robust enough.

**Comment by billsacks** [comment]
@ekluzek I can't think of a reason why it would be beneficial to require `--allow-baseline-overwrite` whenever you're using `--retry`. First, I should have mentioned, in case it isn't clear: with my fix in #3914 , you still need to explicitly add `--allow-baseline-overwrite` if the baselines exist before you run create_test: my fix just adds that implicitly to the retried versions. So in my mind, my fix gives the safety of ensuring that you're not accidentally trying to overwrite pre-existing baselines when running create_test, while still allowing the retried tests to move forward. I can't see a downside here; can you?
